### PR TITLE
[CDAP-6947] allows all rows to be expanded, with or without stack trace

### PR DESCRIPTION
**Pull request by PGuimarais**
expanding row without st

https://issues.cask.co/browse/CDAP-6947

Bamboo Build: 
http://builds.cask.co/browse/CDAP-DRC4203/

### CODE REVIEW

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 1]
```diff
@@ -105,23 +105,28 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
...
-      return;
-    }
 
     if(this.displayData[index].log.stackTrace){
       this.displayData[index].selected = true;
       var stackTraceObj = JSON.parse(JSON.stringify(this.displayData[index]));
       stackTraceObj.stackTrace = true;
       this.displayData.splice(index+1, 0, stackTraceObj);
+    } else {
+      console.log('this element does not have a stack trace - selected: ', this.displayData[index].selected);
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/log-viewer/log-viewer.js]: @PGuimarais please remove.

File: cdap-ui/app/directives/log-viewer/log-viewer.html [hunk 2]
```diff
@@ -114,9 +114,9 @@
              infinite-scroll-container="'.logs-table'"
              >
 
-        <tr ng-class="{'has-stack-trace' : entry.log.stackTrace.length > 0, 'row-selected' : $index+1  < LogViewer.displayData.length && LogViewer.displayData[$index + 1].stackTrace }"
+        <tr ng-class="{'has-stack-trace' : entry.log.stackTrace.length > 0, 'row-selected' : $index+1  < LogViewer.displayData.length && LogViewer.displayData[$index + 1].stackTrace, 'row-expanded' : !LogViewer.displayData[$index].stackTrace && LogViewer.displayData[$index].selected }"
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.html]: LogViewer.displayData[$index] => entry

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 3]
```diff
@@ -105,23 +105,27 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
     });
   };
 
-  this.showStackTrace = (index) => {
-    //If the stack trace is showing, remove it
+  this.toggleStackTrace = (index) => {
+    //If the currently clicked row is a stack trace itself, do nothing
+    if(this.displayData[index].stackTrace && (index - 1) > 0){
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js]: what is the check index-1 is for? that means if the first line has stacktrace, this check will fail?

**Comment by elfenheart** [comment]
👍  LGTM. Squash the commits please

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by elfenheart** [comment]
👍 LGTM

**Pull request merged**